test/cloud/yaml_test.rb

```diff
@@ -0,0 +1,82 @@
+require File.expand_path(File.join(__FILE__, '../..', 'test_helper'))
+require 'rubber/cloud/yaml'
+
+class YAMLTest < Test::Unit::TestCase
+  context 'yaml' do
+    setup do
+      ENV["YAML_DATABASE"] = "/tmp/#{SecureRandom.uuid}"
+
+      env = Rubber::Configuration::Environment::BoundEnv.new({}, nil, nil, nil)
+      @cloud = Rubber::Cloud::YAML.new(env, nil)
+    end
+
+    teardown do
+      FileUtils.rm_f(ENV.delete("YAML_DATABASE"))
+    end
+
+    should 'create instance' do
+      db = [
+        Rubber::Cloud::YAML::Instance.new(SecureRandom.uuid, Rubber::Cloud::YAML::AVAILABLE, 'dc0', '127.0.0.1', '10.0.0.1', nil, nil),
+      ]
+
+      Rubber::Cloud::YAML.persist_database(db, ENV["YAML_DATABASE"])
+
+      assert_equal db.first.id, @cloud.create_instance('', '', '', '', '', '')
+      instance = @cloud.describe_instances(db.first.id).first
+
+      # ensure creating the instance populated the two missing fields.
```

Comment:
From the code above, this seems to only happen when describing the instance, not creating it. Creating it only seems to update the state to ACTIVE.